Core/include/Acts/Detector/interface/IGeometryIdGenerator.hpp

```diff
@@ -0,0 +1,59 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <any>
+
+namespace Acts {
+
+class Surface;
+
+namespace Experimental {
+
+class Portal;
+class DetectorVolume;
+
+/// @brief This is the interface for generating geometry ids and assign them
+/// to detector volumes, portals and surfaces
+///
+class IGeometryIdGenerator {
```

Comment:
Sorry just parachuting in here: this can't reuse/extend the existing `GeometryIdentifierHook` at all?

Reply:
It has to be a replacement: given the feedback we had (also from other experiments) it would be nice if we give the possibility to generate your own GeometryId schema.

The default generator that is implemented shows how this could go (and how it can use a cache for e.g. incremental counting). It would - however, allow a client to fully define that schema (in particular if we bundle it with an include overload of the bitfields).